### ISSUE TITLE
Race condition on Windows freezing window

### ISSUE THREAD

**Issue by DefinitelyNotRobot** [tomaka/winit#633]
This is happening on Windows platform. There seems to be some form of race condition going on that freezes/deadlocks the events loop. Here is the code that causes it:

```
        use winit::*;
        let mut events_loop = winit::EventsLoop::new();
        let window = WindowBuilder::new().build(&events_loop).unwrap();
        std::thread::sleep_ms(20); // Somehow helps to trigger it more often
        window.set_fullscreen(Some(events_loop.get_primary_monitor()));
        window.set_title("test");
        loop {
            events_loop.poll_events(|_| {});
        }
```

**Comment by francesca64** [comment]
This looks like it's the same as #574 or #617

**Comment by francesca64** [comment]
Fixed via #634
